src/app/controllers/reviewGraph.js

```diff
@@ -0,0 +1,1198 @@
+                    }
+                
+                    var ntypes = Array.from(new Set($scope.reviewGraph['nodes'].map(n => calcNodeType(n))));
+                
+                    var processGraph = function(graph) {
+                        for (var n of graph['nodes']) {
+                            var nid = n['identifier'] || n['@id'] || n['url'];
+                            n['id'] = nid;
+                        }
+                        for (var n of graph['nodes']) {
+                            var hlevel = calcNodeHierarchy(n);
+                            n['hierarchyLevel'] = hlevel;
+                        }
+                        for (n of graph['nodes']) {
+                            var nt = calcNodeType(n);
+                            n['group'] = ntypes.indexOf(nt);
+                            n['originalOpacity'] = calcNodeOpacity(n);
+                            n['opacity'] = n['originalOpacity'];
+                            n['nodeSize'] = calcNodeSize(n);
+                            n['nodeScale'] = calcNodeScale(n);
+                            n['enabledNode'] = true
+                        }
+                        for (var l of graph['links']) {
+                            l['value'] = calcLinkValue(l);
+                            l['originalOpacity'] = calcLinkOpacity(l);
+                            l['opacity'] = l['originalOpacity'];
+                        };
+                        graph['mainItemReviewed'] = calcMainItemReviewed();
+                        graph['mainNodeLabel'] = $scope.ciDoc.credibility_label
+                        graph['id'] = $scope.reviewGraph.mainNode
+                        return graph
+                    }
+                
+                    var processedGraph = processGraph($scope.reviewGraph)
+                    if (DEBUG) {
+                        var hlevels = processedGraph['nodes'].filter(n => n.hierarchyLevel != null).map(n => n['hierarchyLevel']);
+                        console.debug('min/max hierarchy levels: ', Math.min(...hlevels), Math.max(...hlevels))
+                    }
+
+                    return processedGraph;
+                }
+
+                // request the corresponding reviewGraph
+                let rg = solrSrv.fetchReviewGraph(doc);
+                rg.success(function(data, status) {
+                    var result = data['results'][0]
+                    $scope.reviewGraph = result['reviewGraph'];
+                    if ($scope.reviewGraph == null) {
+                        alertSrv.set('Warning', 'No review available for this document. Sorry.');
+                    }
+
+                    var processedData = preProcessedGraph($scope.reviewGraph)
+                    if (DEBUG) {console.debug('reviewGraph data: ', processedData)}
+
+                    $scope.graph = processedData
+
+                    // trigger rendering of the graph
+                    $scope.renderGraph($scope.graph);
+                });
```

Comment:
Nice an clear: fetch, pre-process and render.

I'd rename `preProcessedGraph` to `preProcessGraph` or, longer but more descriptive `acred_to_d3_ReviewGraph`

No need to fix this yet, but to make it even cleaner I'd consider moving the whole definition of `preProcessedGraph` one level up? I'm a bit concerned about changes to and calls to  `$scope` within that function.

Reply:
Ok, changed the _preProcessedGraph_ name to _acred_to_d3_ReviewGraph_ and added a TODO for moving this process one level up